docs/architecture/transactions/kernel.md

```diff
@@ -102,6 +102,10 @@ For every note, the [MAST root](https://0xpolygonmiden.github.io/miden-vm/design
         exec.note::prepare_note
         # => [NOTE_SCRIPT_HASH]
 
+        # store the NOTE_SCRIPT_HASH to the memory to be able to call it dynamically
+        loc_storew.0 dropw locaddr.0
+        # => [note_script_hash_addr]
```

Comment:
Could `notes::perpare_note` return the address of script root rather than the script root itself? This way, we don't have store it into memory again, right?